### PR TITLE
Replace food_recipe.json heating inlined lists with requirements

### PR DESCRIPTION
**Pull request by Coolthulhu**
The total diff line looks scary, but the additions are around 300 so it's perfectly readable. Plus, vast majority of the changes are almost trivial in nature.
Almost 1.5k lines of repeated boilerplate json purged in one of the biggest cleanups in the history of the repo.

This also fixes some bad recipes where charcoal smoker was unusable for some recipes it should be usable for. Any recipe that allows both fire and hotplate must also allow charcoal smoker unless there is a good reason for why shouldn't it be the case (I can't think of any).
Recipes can allow only hotplate and no fire if they require precise heating or fire and no hotplate if it requires open flame, but if it allows both, it allows charcoal smoker.

Some water recipes also allow coffeemaker. Those got a different list.

### CODE REVIEW

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -3403,14 +3397,8 @@
...
   ], "tools": [
-    [
-      [ "thermometer", -1 ]
-    ],
-    [
-      [ "hotplate", 8 ],
-      [ "toolset", 8 ],
-      [ "fire", -1 ]
-    ]
+    [ [ "surface_heat", 12, "LIST" ] ],
```
Comment (kolsurma) [inline data/json/recipes/recipe_food.json]: copy paste error, should be `"surface_heat", 8, "LIST"`

File: data/json/recipes/recipe_food.json [hunk 2]
```diff
@@ -2343,18 +2277,8 @@
...
-      [ "hotplate", 2 ],
-      [ "char_smoker", 2 ],
-      [ "toolset", 2 ],
-      [ "atomic_coffeepot", -1 ],
-      [ "coffeemaker", 2],
-      [ "fire", -1 ]
-    ]
-  ],
+  "qualities":[ {"id":"BOIL","level":1} ],
+  "tools": [ [ [ "water_boiling_heat", 3, "LIST" ] ] ],
```
Comment (kolsurma) [inline data/json/recipes/recipe_food.json]: should be `"water_boiling_heat", 2, "LIST"`

Comment (Coolthulhu) [inline data/json/recipes/recipe_food.json]: 2 is for teas.
If we're talking balance/realism, it should be 10 or so.

Comment (kolsurma) [inline data/json/recipes/recipe_food.json]: Eh, I was just pointing out that it was previously 2. If it should be changed to something more, that's fine. Should it be done in this PR though?

Comment (Coolthulhu) [inline data/json/recipes/recipe_food.json]: Looks like you got the wrong line.

I think I replaced one 2 with 3 for consistency, but it was somewhere else.

Comment (Coolthulhu) [inline data/json/recipes/recipe_food.json]: Or it may be github which got the line wrong - it looks different in the email quote.

Comment (kolsurma) [inline data/json/recipes/recipe_food.json]: I wouldn't doubt it. Thank you github for making this weird.

```
"type" : "recipe",
   "result": "coffee",
   "category": "CC_FOOD",
   "subcategory": "CSC_FOOD_DRINKS",
   "skill_used": "cooking",
    "time": 10000,
    "autolearn": true,
    "batch_time_factors":[80, 4],
 -  "qualities":[
 -    {"id":"BOIL","level":1}
 -  ], "tools": [
 -    [
 -      [ "hotplate", 2 ],
 -      [ "char_smoker", 2 ],
 -      [ "toolset", 2 ],
 -      [ "atomic_coffeepot", -1 ],
 -      [ "coffeemaker", 2],
 -      [ "fire", -1 ]
 -    ]
 -  ],
 +  "qualities":[ {"id":"BOIL","level":1} ],
 +  "tools": [ [ [ "water_boiling_heat", 3, "LIST" ] ] ],
```

Comment (kolsurma) [inline data/json/recipes/recipe_food.json]: That said, leave it at 3, it makes sense that it would take as much energy as boiling water.

**Pull request merged**